cocotb/__init__.py

```diff
@@ -254,7 +254,7 @@ def _initialise_testbench_(argv_):
     # sys.path normally includes "" (the current directory), but does not appear to when python is embedded.
     # Add it back because users expect to be able to import files in their test directory.
     # TODO: move this to gpi_embed.cpp
-    sys.path.insert(0, "")
+    sys.path.append("")
```

Comment:
I'm positive this is not a valid change. We want local modules to be resolved before external ones. If there are collisions because of this, we need to put the power in the user's hand to solve. A better solution is to not do this at all. While "users expect" things, all of a sudden tests not working and the user not immediately knowing why or having the power to change that behavior are bad.

Reply:
The intention for this change is to smoke out issues that may be related to PATH, as I suspected prepending things to PATH may be part of the issue. If the CI gets fixed I can revert this change.